pandas/core/window/rolling.py

```diff
@@ -93,6 +93,18 @@ def __init__(
         self.axis = obj._get_axis_number(axis) if axis is not None else None
         self.validate()
         self._numba_func_cache: Dict[Optional[str], Callable] = dict()
+        self.exclusions = kwargs.get("exclusions", set())
+
+    def _shallow_copy(self, obj: FrameOrSeries, **kwargs) -> ShallowMixin:
+        if isinstance(obj, ABCDataFrame):
```

Comment:
WAT? i would much rather properly copy in the super class. really avoid if/then cases unless absolutely necessary.

Reply:
IOW we can *always* just copy exclusions

Reply:
@jreback sure, I can remove the if-then cases here, but then I need to potentially remove elements from `self.exclusions` in
```
def _obj_with_exclusions(self)
```
, otherwise this could throw an error if there are elements of `self.exclusions` that are no longer part of `self.obj.columns`:
```
return self.obj.drop(self.exclusions, axis=1)
```

Reply:
Anyway, that's what I've done in the latest commit